pkg/controller/gc_controller_test.go

```diff
@@ -241,6 +247,208 @@ func TestGarbageCollect(t *testing.T) {
 	}
 }
 
+func TestGarbageCollectBackup(t *testing.T) {
+	tests := []struct {
+		name                   string
+		backup                 *api.Backup
+		deleteBackupFile       bool
+		snapshots              sets.String
+		backupFiles            sets.String
+		backupMetadataFiles    sets.String
+		restores               []*api.Restore
+		expectedRestoreDeletes []string
+		expectedBackupDelete   string
+		expectedSnapshots      sets.String
+		expectedBackupFiles    sets.String
+		expectedMetadataFiles  sets.String
+	}{
+		{
+			name: "failed snapshot deletion",
```

Comment:
could you include what the expected outcome is in the name for all cases, e.g. something like "failed snapshot deletion shouldn't delete backup"?